src/view/reviewManager.ts

```diff
@@ -641,260 +644,28 @@ export class ReviewManager {
 		return selected;
 	}
 
-	private async getPullRequestTitleAndDescriptionDefaults(progress: vscode.Progress<{ message?: string, increment?: number }>, pullRequestDescriptionMethod: PullRequestDescriptionSource): Promise<{ title: string, description: string } | undefined> {
-		let template: vscode.Uri | undefined;
+	public async createPullRequest(isDraft?: boolean): Promise<void> {
+		vscode.commands.executeCommand('setContext', 'github:createPullRequest', true);
+		if (!this._createPRViewProvider) {
+			this._createPRViewProvider = new CreatePullRequestViewProvider(this._context.extensionUri, this._folderRepoManager, !!isDraft);
 
-		// Only fetch pull request templates if requested
-		if (pullRequestDescriptionMethod === PullRequestDescriptionSourceEnum.Template) {
-			const pullRequestTemplates = await this._folderRepoManager.getPullRequestTemplates();
+			this._createPRViewDisposables.push(this._createPRViewProvider.onDone(async createdPR => {
+				vscode.commands.executeCommand('setContext', 'github:createPullRequest', false);
 
-			if (pullRequestTemplates.length === 1) {
-				template = pullRequestTemplates[0];
-				progress.report({ increment: 5, message: 'Found pull request template. Creating pull request...' });
-			}
+				this._createPRViewProvider?.dispose();
+				this._createPRViewDisposables.forEach(d => d.dispose());
+				this._createPRViewProvider = undefined;
 
-			if (pullRequestTemplates.length > 1) {
-				const targetTemplate = await vscode.window.showQuickPick(pullRequestTemplates.map(uri => {
-					return {
-						label: vscode.workspace.asRelativePath(uri.path),
-						uri: uri
-					};
-				}), {
-					ignoreFocusOut: true,
-					placeHolder: 'Select the pull request template to use'
-				});
-
-				// Treat user pressing escape as cancel
-				if (!targetTemplate) {
-					return;
+				if (createdPR) {
+					await this.updateState();
+					await vscode.commands.executeCommand('pr.openDescription', createdPR);
```

Comment:
Totally up to you, but I find these `executeCommand`s make it difficult to refactor and navigate the code. Instead, you could refactor the `pr.openDescription` command to call a function, then call that same function here.

Reply:
Yeah, good point, the lack of type checking for the arguments is also a major downside. I've factored out a function to use